### PR TITLE
TemplateInstance metrics update

### PR DESCRIPTION
**Pull request by bparees**
tweak of https://github.com/openshift/origin/pull/18650

### CODE REVIEW

**Comment by bparees** [comment]
@gabemontero ptal

File: pkg/template/controller/metrics.go [hunk 1]
```diff
@@ -1,76 +1,79 @@
...
+	templateInstanceActiveAge := newTemplateInstanceActiveAge()
+
+	templateInstanceCompleted.Describe(ch)
+	templateInstanceActiveAge.Describe(ch)
 }
 
 func (c *TemplateInstanceController) Collect(ch chan<- prometheus.Metric) {
+	templateInstanceCompleted.Collect(ch)
+
+	now := c.clock.Now()
```
Comment (gabemontero) [inline pkg/template/controller/metrics.go]: I forget, was there a reason that unix epoch time was not used here?

Curious on the req here vs. what we had to the for build's active metric

File: pkg/template/controller/metrics_test.go [hunk 2]
```diff
@@ -41,58 +60,132 @@ func (f *fakeResponseWriter) WriteHeader(statusCode int) {
...
+openshift_template_instance_active_age_seconds_bucket{le="+Inf"} 1
+openshift_template_instance_active_age_seconds_sum 900
+openshift_template_instance_active_age_seconds_count 1
+# HELP openshift_template_instance_completed_total Counts completed TemplateInstance objects by condition
+# TYPE openshift_template_instance_completed_total counter
+openshift_template_instance_completed_total{condition="InstantiateFailure"} 2
+openshift_template_instance_completed_total{condition="Ready"} 1
 `
+
+	clock := &fakeClock{now: time.Unix(0, 0)}
```
Comment (gabemontero) [inline pkg/template/controller/metrics_test.go]: Yeah here we are using unix epoch ...

File: pkg/template/controller/metrics.go (hunk 1 above)
Comment (bparees) [reply pkg/template/controller/metrics.go]: In this case, c.clock will be an instance of RealClock which has as its Now implementation:

```
// Now returns the current time.
func (RealClock) Now() time.Time {
	return time.Now()
}
```

File: pkg/template/controller/templateinstance_controller.go [hunk 3]
```diff
@@ -80,6 +83,7 @@ func NewTemplateInstanceController(config *rest.Config, kc kclientsetinternal.In
 		informer:         informer.Informer(),
 		queue:            workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "openshift_template_instance_controller"),
 		readinessLimiter: workqueue.NewItemFastSlowRateLimiter(5*time.Second, 20*time.Second, 200),
+		clock:            clock.RealClock{},
```
Comment (bparees) [inline pkg/template/controller/templateinstance_controller.go]: @gabemontero ^^

File: pkg/template/controller/metrics.go (hunk 1 above)
Comment (bparees) [reply pkg/template/controller/metrics.go]: (it was done this way to provide an abstraction for the clock implementation so tests could plug in their own clock impl)

**Comment by bparees** [comment]
@adambkaplan mildly relevant to your interests.

**Pull request merged**